src/pages/ReimbursementAccount/NonUSD/BusinessInfo/substeps/AverageReimbursement.tsx

```diff
@@ -0,0 +1,80 @@
+
+type AverageReimbursementProps = SubStepProps;
+
+const {TRADE_VOLUME} = INPUT_IDS.ADDITIONAL_DATA.CORPAY;
+const STEP_FIELDS = [TRADE_VOLUME];
+
+function AverageReimbursement({onNext, isEditing}: AverageReimbursementProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [reimbursementAccount] = useOnyx(ONYXKEYS.REIMBURSEMENT_ACCOUNT);
+    const [reimbursementAccountDraft] = useOnyx(ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM_DRAFT);
+    const [corpayOnboardingFields] = useOnyx(ONYXKEYS.CORPAY_ONBOARDING_FIELDS);
+    const policyID = reimbursementAccount?.achData?.policyID;
+    const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`);
+    const currency = policy?.outputCurrency ?? '';
+
+    const tradeVolumeRangeListOptions = useMemo(() => {
+        if (!corpayOnboardingFields?.picklists.TradeVolumeRange) {
+            return {};
+        }
+
+        return corpayOnboardingFields.picklists.TradeVolumeRange.reduce((accumulator, currentValue) => {
+            accumulator[currentValue.name] = currentValue.stringValue;
+            return accumulator;
+        }, {} as Record<string, string>);
+    }, [corpayOnboardingFields]);
+
+    const tradeVolumeDefaultValue = reimbursementAccount?.achData?.additionalData?.corpay?.[TRADE_VOLUME] ?? reimbursementAccountDraft?.[TRADE_VOLUME] ?? '';
+
+    const validate = useCallback((values: FormOnyxValues<typeof ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM> => {
+        return ValidationUtils.getFieldRequiredErrors(values, STEP_FIELDS);
+    }, []);
+
+    const handleSubmit = useReimbursementAccountStepFormSubmit({
+        fieldIds: STEP_FIELDS,
+        onNext,
+        shouldSaveDraft: isEditing,
+    });
+
+    return (
+        <FormProvider
+            formID={ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM}
+            submitButtonText={translate(isEditing ? 'common.confirm' : 'common.next')}
+            onSubmit={handleSubmit}
+            validate={validate}
+            style={[styles.flexGrow1]}
+            submitButtonStyles={[styles.mh5]}
+        >
+            <Text style={[styles.textHeadlineLineHeightXXL, styles.mh5, styles.mb3]}>{translate('businessInfoStep.whatsYourExpectedAverageReimbursements')}</Text>
+            <InputWrapper
+                InputComponent={PushRowWithModal}
+                optionsList={tradeVolumeRangeListOptions}
+                description={translate('businessInfoStep.averageReimbursementAmountInCurrency', {currencyCode: currency})}
+                modalHeaderTitle={translate('businessInfoStep.selectAverageReimbursement')}
+                searchInputTitle={translate('businessInfoStep.findAverageReimbursement')}
+                inputID={TRADE_VOLUME}
+                shouldSaveDraft={!isEditing}
+                defaultValue={tradeVolumeDefaultValue}
+            />
+        </FormProvider>
+    );
```

Comment:
Looks like we have `SingleFieldStep` for this kind of Form. Can we reuse it instead?

Reply:
SingleFieldStep is used for TextInput fields. Here we have a different component. Maybe in the future we will rework SingleFieldStep so its more flexible or create a different reusable component for PushRowWithModal case as we have a few of these